File: Coding/Nicomachus Theorem/Squared_triangular_number3.py
# Efficient Python program to find sum of cubes
# of first n natural numbers that avoids
# overflow if result is going to be with in
# limits.

# A000537 		Sum of first n cubes; or n-th triangular number squared. 

# Returns the sum of series
def sumOfSeries(n):
	x = 0
	if n % 2 == 0 :
		x = (n / 2) * (n + 1)
	else:
		x = ((n + 1) / 2) * n
		
	return (int)(x * x)


# Driver Function
n = int(input())
print(sumOfSeries(n))

# Code Contributed by Mohit Gupta_OMG <(0_o)>

File: Coding/Nicomachus Theorem/Squared_triangular_number3.cpp
// Efficient CPP program to find sum of cubes
// of first n natural numbers that avoids
// overflow if result is going to be with in
// limits.

// A000537 		Sum of first n cubes; or n-th triangular number squared. 
#include <iostream>
using namespace std;

// Returns sum of first n natural
// numbers
int sumOfSeries(int n)
{
	int x;
	if (n % 2 == 0)
		x = (n / 2) * (n + 1);
	else
		x = ((n + 1) / 2) * n;
	return x * x;
}

// Driver code
int main()
{
	int n =0;
	cin >> n;
	cout << sumOfSeries(n) << endl;
	return 0;
}
